lib/by_patron_type_factory.js

```diff
@@ -2,18 +2,31 @@ const jsonldParseUtils = require('./jsonld-parse-utils')
 const FactoryBase = require('./factory_base')
 
 class ByPatronTypeFactory extends FactoryBase {
-  // returns a mapping with Sierra codes at its top-level
+  // returns a mapping with ptype numbers at its top-level
   static createMapping () {
     this.patronTypeJSON = this._getPatronTypeJsonLD()['@graph']
+    this.sierraLocationsJSON = this._getSierraJsonLD()['@graph']
     const result = {}
-
     for (const index in this.patronTypeJSON) {
       const accessibleDeliveryLocationTypes = jsonldParseUtils.forcetoFlatArray(this.patronTypeJSON[index]['nypl:deliveryLocationAccess'])
 
+      const scholarRoomCode = (this.patronTypeJSON[index]['nypl:scholarRoom'])
+      let scholarRoomLabel
+      if (scholarRoomCode) {
+        scholarRoomLabel = this.sierraLocationsJSON
+          .find((location) => location['@id'] === scholarRoomCode['@id'])['skos:prefLabel']
+      }
+
       result[this.patronTypeJSON[index]['skos:notation']] = {
         label: this.patronTypeJSON[index]['skos:prefLabel'],
         accessibleDeliveryLocationTypes: accessibleDeliveryLocationTypes
       }
+      if (scholarRoomCode && scholarRoomLabel) {
+        result[this.patronTypeJSON[index]['skos:notation']].scholarRoom = {
+          code: scholarRoomCode,
```

Comment:
I believe this `scholarRoomCode` is actually the full entity, e.g.
```
code: { '@id': 'nyplLocation:mal17' }
```
Whereas ideally it's just 
```
code: 'mal17'`
```